currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -201,14 +198,20 @@ def _update(self, date_from, date_to, newest_only=False):
                         )
 
             if is_scheduled:
+                provider._schedule_last_successful_run()
                 provider._schedule_next_run()
 
+    def _schedule_last_successful_run(self):
+        self.last_successful_run = self.next_run
+
     def _schedule_next_run(self):
         self.ensure_one()
-        self.last_successful_run = self.next_run
-        self.next_run = (
-            datetime.combine(self.next_run, time.min) + self._get_next_run_period()
-        ).date()
+        self.next_run = min(
+            (
+                datetime.combine(self.next_run, time.min) + self._get_next_run_period()
+            ).date(),
+            fields.Date.today(),
+        )
```

Comment:
Sorry, I am not getting the rationale of this change. Doesn't this practically revoke the scheduled update interval settings by always setting today's date?

Reply:
I think this explanation may help to understand it: https://github.com/OCA/currency/pull/147#issuecomment-1346322636

Reply:
@victoralmau I read it and I think this logic fails on the pattern 2 - `next_run` does not get updated with this logic if I'm not mistaken.